tick/stateful/functions_test.go

```diff
@@ -485,3 +486,63 @@ func Test_Duration(t *testing.T) {
 		}
 	}
 }
+
+func Test_If(t *testing.T) {
+	f := &ifFunc{}
+
+	testCases := []struct {
+		args []interface{}
+		exp  interface{}
+		err  error
+	}{
+		// Error cases
+		{
+			args: []interface{}{true},
+			err:  errors.New("if expects exactly three argument"),
+		},
+		{
+			args: []interface{}{true, 6},
+			err:  errors.New("if expects exactly three argument"),
+		},
+		{
+			args: []interface{}{12, 6, false},
+			err:  errors.New("first argument to if must be a condition with type of bool - got int"),
+		},
+
+		// Simple if
+		{
+			args: []interface{}{true, 1, 2},
+			exp:  1,
+		},
+		{
+			args: []interface{}{false, 1, 2},
+			exp:  2,
+		},
+
+		// multiple types
+		{
+			args: []interface{}{false, 1, "1"},
```

Comment:
Do we want to support this? I can't think of a valid use case right now and it could lead to confusing results later on.

For example:

```
|eval(lambda: if("value", 1, 2.5))
```

The type would alternate between int and float. Which would be very confusing, and hard to debug.

What if we added a check that enforced that the two last args are the same type? Then we could catch the error sooner and help the user fix the mistake. Thoughts?